src/components/MapComp/index.tsx

```diff
@@ -1,86 +1,63 @@
-import React, { useEffect, useMemo, useRef, useState } from 'react';
+import React, { useEffect, useMemo, useRef } from 'react';
 import { useRecoilValue } from 'recoil';
 import { css } from '@emotion/react';
 import PointMarker from '@components/Marker/pointMarker';
+import ClusterMarker from '@components/Marker/clusterMarker';
 import { CustomGeoJSONFeatures } from '@libs/types/map';
 import { renderEmotionElementToHtml } from '@libs/utils/renderEmotionElementToHtml';
 import { geoJsonSelector } from '@states/map';
 import { activeFilterIdAtom } from '@states/ui';
-import mapboxgl, { GeoJSONSourceRaw, Marker } from 'mapbox-gl';
-import { FeatureCollection } from 'geojson';
+import mapboxgl, { MapboxGeoJSONFeature, Marker } from 'mapbox-gl';
 import 'mapbox-gl/dist/mapbox-gl.css';
+import { FilterId } from '@libs/types/filter';
+import { FeatureCollection } from 'geojson';
 import { mapConfig } from './utils/mapConfig';
 
 type MapCompProps = {
   handleClickMarker: (id: number) => void;
 };
 
 function MapComp({ handleClickMarker }: MapCompProps) {
-  const features = useRecoilValue(geoJsonSelector);
-  const [featuresOnScreen, setFeaturesOnScreen] = useState<CustomGeoJSONFeatures[]>();
   const activeFilterId = useRecoilValue(activeFilterIdAtom);
+  const allFeatures = useRecoilValue(geoJsonSelector);
   const mapRef = useRef<mapboxgl.Map>();
-  const markerList: { [id in number | string]: mapboxgl.Marker } = useMemo(() => ({}), []);
+  const activeFilterIdRef = useRef(activeFilterId);
+  const pointMarkerList: { [id in number | string]: Marker } = useMemo(() => ({}), []);
+  const clusterMarkerList: Marker[] = useMemo(() => [], []);
 
-  useEffect(() => {
-    mapboxgl.accessToken = `${process.env.REACT_APP_MAPBOX_ACCESS_TOKKEN}`;
-    mapRef.current = new mapboxgl.Map(mapConfig);
+  const updateMarkers = () => {
     const map = mapRef.current;
+    const filterId = activeFilterIdRef.current;
 
-    const updateFeaturesOnScreen = () => {
-      const mapboxFeaturesOnScreen = map.querySourceFeatures('placeList');
-      const uniqueIds = new Set<number>();
-
-      mapboxFeaturesOnScreen.forEach((feature) => {
+    if (!map) return;
+    const mapboxFeaturesOnScreen = map.querySourceFeatures(`cafeList/${filterId}`);
+    const clustersOnScreen: MapboxGeoJSONFeature[] = [];
+    const uniqueClusterIds = new Set<number>();
+    const uniquePointIds = new Set<number>();
+    mapboxFeaturesOnScreen.forEach((feature) => {
+      if (feature.properties?.cluster) {
```

Comment:
if 중첩문을 플랫하게 바꾸면 좀 더 보기 쉬울것같아요~~

```suggestion
      if (feature.properties?.cluster && uniqueClusterIds.has(feature.properties.cluster.id) return;
      
      if(feature.proprties?.cluster){
           clusterOnScreen.push(feature)
      }
      
      if(feature.properties?.id ){
          uniquePointIds.add(feature.properties?.id)
      }
      ```
      이런식으로 뎁스를 없애보면 어떨까요?